mrmustard/lab_dev/wires.py

```diff
@@ -336,6 +336,14 @@ def __bool__(self) -> bool:
         r"Returns ``True`` if this ``Wires`` object has any wires, ``False`` otherwise."
         return any(self.args)
 
+    def __len__(self) -> int:
+        r"The number of wires."
+        try:
+            return self._len
+        except AttributeError:
+            self._len = sum(map(len, self.args))
+            return self._len
```

Comment:
is this expensive/does it need to be cached?

Reply:
not expensive, but also will never change so might as well cache it.